in_toto/in_toto_run.py

```diff
@@ -102,37 +99,53 @@ def main():
                " --key <functionary private key path>\n{0}"
                "[--materials <filepath>[ <filepath> ...]]\n{0}"
                "[--products <filepath>[ <filepath> ...]]\n{0}"
-               "[--record-byproducts] -- <cmd> [args]\n\n"
+               "[--record-byproducts]\n{0}"
+               "[--verbose] -- <cmd> [args]\n\n"
                .format(lpad))
 
   in_toto_args = parser.add_argument_group("in-toto options")
-  # FIXME: Name has to be unique!!! Where will we check this?
+
   # FIXME: Do we limit the allowed characters for the name?
   in_toto_args.add_argument("-n", "--step-name", type=str, required=True,
       help="Unique name for link metadata")
 
   in_toto_args.add_argument("-m", "--materials", type=str, required=False,
       nargs='+', help="Files to record before link command execution")
+
   in_toto_args.add_argument("-p", "--products", type=str, required=False,
       nargs='+', help="Files to record after link command execution")
 
   in_toto_args.add_argument("-k", "--key", type=str, required=True,
       help="Path to private key to sign link metadata (PEM)")
 
-  in_toto_args.add_argument("-b", "--record-byproducts", dest='record_byproducts',
+  in_toto_args.add_argument("-b", "--record-byproducts",
       help="If set redirects stdout/stderr and stores to link metadata",
-      default=False, action='store_true')
+      dest="record_byproducts", default=False, action="store_true")
+
+  in_toto_args.add_argument("-v", "--verbose", dest="verbose",
+      help="Verbose execution.", default=False, action="store_true")
 
-  link_args = parser.add_argument_group("Link command")
   # FIXME: This is not yet ideal.
   # What should we do with tokens like > or ; ?
-  link_args.add_argument("link_cmd", nargs="+",
+  in_toto_args.add_argument("link_cmd", nargs="+",
     help="Link command to be executed with options and arguments")
 
   args = parser.parse_args()
 
-  in_toto_run(args.step_name, args.key, args.materials, args.products,
-      args.link_cmd, args.record_byproducts)
+  # Turn on all the `log.doing()` in the library
+  if args.verbose:
+    log.logging.getLogger().setLevel(log.logging.INFO)
+
+  # We load the key here because it might prompt the user for a password in
+  # case the key is encrypted. Something that should not happen in the library.
+  try:
+    key = util.prompt_import_rsa_key_from_file(args.key)
+  except Exception as e:
+    log.error("in load key - {}".format(args.key))
+    sys.exit(1)
+
+  in_toto_run(args.step_name, args.materials, args.products,
+      args.link_cmd, key, args.record_byproducts)
 
-if __name__ == '__main__':
+if __name__ == "__main__":
```

Comment:
Well, right now the logger does not have file handler. And it is on us to decide how the logger is used. :)

I'd like to improve logging/user feedback in general (cf. https://github.com/in-toto/in-toto/issues/6). E.g. properly separate logging (which could use any handler) and user feedback (which must print to the terminal). We have discussed this before and decided that python's `logging` module can be used for both (given proper namespacing).